src/lib/y2storage/filesystems/blk_filesystem.rb

```diff
@@ -121,6 +121,35 @@ def top_level_btrfs_subvolume
         BtrfsSubvolume.new(storage_subvol)
       end
 
+      # The default Btrfs subvolume (typically @).
+      #
+      # @note When a new Btrfs filesystem is created, a top level subvolume is added and
+      #   initialized as default subvolume.
+      #
+      # @see #top_level_btrfs_subvolume
+      #
+      # @return [BtrfsSubvolume] nil if it makes no sense for this filesystem
+      def default_btrfs_subvolume
+        return nil unless supports_btrfs_subvolumes?
+
+        storage_subvol = Storage.to_btrfs(to_storage_value).default_btrfs_subvolume
+        BtrfsSubvolume.new(storage_subvol)
+      end
+
+      # Finds a subvolume by its path
+      #
+      # @param path [String] subvolume path
+      # @return [BtrfsSubvolume, nil] nil if it does not find a subvolume with this path
+      def find_btrfs_subvolume_by_path(path)
+        return nil unless supports_btrfs_subvolumes?
+
+        btrfs = Storage.to_btrfs(to_storage_value)
+        storage_subvol = btrfs.find_btrfs_subvolume_by_path(path)
+        BtrfsSubvolume.new(storage_subvol)
+      rescue Storage::BtrfsSubvolumeNotFoundByPath
```

Comment:
Whatever we do, I think this PBI is the right time to address it. Note that "this PBI" doesn't imply this PR. I'm fine with merging as it is and fixing the subclassing issue in a separate PR inside the same PBI.

Back to the subject, IMO we need to subclass because there is too much Btrfs specific code here now. The question is whether that should imply that we need to create a wrapper for each possible `BlkFilesystem` subclass (that is, we must create all the sibling classes for `Btrfs`). As I see it, we have two options:

- Create only the `Btrfs < BlkFilesystem` class, but no other subclasses. That's simpler and makes sense since `Btrfs` is the only block filesystem with an extended API. All the other `BlkFilesystem` subclasses in libstorage simply adhere to the parent class API, without offering any new method (so they don't deserve to be separate classes on their own). This approach have a problem, that we have a class that have one subclass (`Btrfs < BlkFilesystem`) but it's not abstract. That could be considered an "API smell".

- Another solution would be to create the following 9 classes as subclasses of block filesystem - `Btrfs`, `Ext`, `Iso9660`, `Ntfs`, `Reiserfs`, `Swap`, `Udf`, `Vfat` and `Xfs`. That's closer to libstorage-ng (which is good and bad). The point is that 8 of those 9 classes offer absolutely nothing else than the parent `BlkFilesystem` class, so they are irrelevant from the API point of view and they just need maintenance job to yast-storage-ng (we need to add just another empty subclass for each newly supported filesystem).

So we have to choose one API smell here: having a subclass with no siblings (solution 1) or having many subclasses that are only different in the implementation but not from the API point of view (solution 2).